lib/countly.js

```diff
@@ -3316,6 +3368,53 @@
         return ob;
     }
 
+    /**
+     * Truncates an object's key/value pairs to a certain length
+     * @param {Object} obj - original object to be truncated
+     * @param {Number} keyLimit - limit for key length
+     * @param {Number} valueLimit - limit for value length
+     * @returns {Object} - the new truncated object
+     */
+    function truncateObject(obj, keyLimit, valueLimit) {
+        var ob = {};
+        if (Object.keys(obj).length > this.maxSegmentationValues) {
+            var resizedObj = {};
+            for (var e in obj) {
+                resizedObj[e] = obj[e];
+            }
+            obj = resizedObj;
+        }
+        for (var key in obj) {
+            var newKey = key;
+            var newValue = obj[key];
+            if (key.length > keyLimit) {
+                newKey = key.substring(0, keyLimit);
+                this.log("Key: " + key + " is longer than accepted length. It will be truncated.");
+            }
+            if (obj[key].length > valueLimit && (typeof obj[key] === 'string' || typeof obj[key] === 'number')) {
+                newValue = obj[key].substring(0, valueLimit);
+                this.log("Value: " + obj[key] + " is longer than accepted length. It will be truncated.");
+            }
+            ob[newKey] = newValue;
+        }
+        return ob;
+    }
+
+    /**
+     * Truncates a single string to a certain length
+     * @param {string|number} str - original string to be truncated
+     * @param {Number} limit - limit length
+     * @returns {string|number} - the new truncated string
+     */
+    function truncateSingleValue(str, limit) {
+        var newStr = str;
+        if (str.length > limit && (typeof str === 'string' || typeof str === 'number')) {
```

Comment:
Type check would probably need to come first. I think it would throw an exception if you would call ".length" on a number or boolean.